modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2SynapseEnvironment.java

```diff
@@ -334,6 +324,16 @@ public boolean injectInbound(final MessageContext synCtx, SequenceMediator seq,
                     + (seq.getName() == null ? "Anonymous" : seq.getName()) + " Sequence");
         }
 
+        Collection<InboundEndpoint> inboundEndpoints = synCtx.getConfiguration().getInboundEndpoints();
+
+        if (synCtx.getProperty("car.deployed.name") != null) {
+            for (InboundEndpoint inboundEndpoint : inboundEndpoints) {
+                if (synCtx.getProperty("car.deployed.name").equals(inboundEndpoint.getName())) {
```

Comment:
Can you please explain this logic. How the car name is equal to the inbound endpoint name ?